test/threadstest.c

```diff
@@ -473,6 +473,34 @@ static int test_multi(int idx)
     return testresult;
 }
 
+/*
+ * This test attempts to load several providers at the same time, and if
+ * run with a thread sanitizer, should crash if the core provider code
+ * doesn't synchronize well enough.
+ */
+#define MULTI_LOAD_THREADS 3
```

Comment:
How reliably does this reproduce problems?
Would more threads be more likely to produce the problem or not?
This looks like a _cheap_ test.

For reference: two threads seemed like enough when I was testing.

Reply:
Two threads seemed fine in my manual tests as well, but I increased that a bit for good measure.  Other than that, I frankly have no idea, but am thinking that if we need to update this, we can do that *later*, i.e. doesn't have to be done in this PR.